apps/rush-lib/src/utilities/VersionControl.ts

```diff
@@ -41,6 +41,41 @@ export default class VersionControl {
     });
   }
 
+  public static getRemoteMasterBranch(repositoryUrl?: string): string | undefined {
+    const defaultRemote: string = 'origin';
+    const defaultMaster: string = 'origin/master';
+    let useDefault: boolean = false;
+    let matchingRemotes: string[] = [];
+
+    if (!repositoryUrl) {
+      useDefault = true;
+    } else {
+      const output: string = child_process
+      .execSync(`git remote`)
+      .toString();
+      matchingRemotes = output.split('\n').filter(remoteName => {
+        if (remoteName) {
+          const remoteUrl: string = child_process.execSync(`git remote get-url ${remoteName}`)
+            .toString()
+            .trim();
+          if (remoteName === defaultRemote && remoteUrl === repositoryUrl) {
```

Comment:
It is safer to compare url in a case-sensitive way.

---
In reply to: [134301332](https://github.com/Microsoft/web-build-tools/pull/328#discussion_r134301332) [](ancestors = 134301332)